src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -73,6 +80,9 @@ public class SftpClient {
   private static final Logger LOGGER = getLogger(SftpClient.class);
   protected static final OpenMode[] CREATE_MODES = {OpenMode.Write, OpenMode.Create};
   protected static final OpenMode[] APPEND_MODES = {OpenMode.Write, OpenMode.Append};
+  public static final Long PWD_COMMAND_EXECUTION_TIMEOUT = 2L;
```

Comment:
Seems like a too low timeout value. It could fail without being the case you are trying to solve. Consider using at least 15 to 30 seconds.